plugins/auth-backend/src/identity/TokenFactory.ts

```diff
@@ -127,23 +128,28 @@ export class TokenFactory implements TokenIssuer {
     this.keyExpiry = moment().add(this.keyDurationSeconds, 'seconds');
     const promise = (async () => {
       // This generates a new signing key to be used to sign tokens until the next key rotation
-      const key = await JWK.generate('EC', 'P-256', {
-        use: 'sig',
-        kid: uuid(),
-        alg: 'ES256',
-      });
+
+      const key = await generateSecret('HS256');
+      const kid = uuid();
+      const jwk = await fromKeyLike(key);
+      jwk.kid = kid;
+      jwk.alg = 'HS256';
 
       // We're not allowed to use the key until it has been successfully stored
       // TODO: some token verification implementations aggressively cache the list of keys, and
       //       don't attempt to fetch new ones even if they encounter an unknown kid. Therefore we
       //       may want to keep using the existing key for some period of time until we switch to
       //       the new one. This also needs to be implemented cross-service though, meaning new services
       //       that boot up need to be able to grab an existing key to use for signing.
-      this.logger.info(`Created new signing key ${key.kid}`);
-      await this.keyStore.addKey((key.toJWK(false) as unknown) as AnyJWK);
+      this.logger.info(`Created new signing key ${jwk.kid}`);
+      await this.keyStore.addKey({
+        // @ts-ignore
```

Comment:
I reworked this a bit and added a comment in the code about why this is necessary. Basically, TS doesn't differentiate between a missing property (no key) and a key that has an undefined value. So TSC is complaining here that it can't map this object to type AnyJWT (which extends from type Record<String, String>), because it can't validate that all keys have values of type string. In practice, the JOSE library is only setting keys/values on fields that it returns, and is not setting explicit "undefined" for these optional fields - but again, we have no way in TS to differentiate the two. 

https://github.com/microsoft/TypeScript/issues/13195